logstreams/src/test/java/io/zeebe/logstreams/processor/AsyncSnapshotingTest.java

```diff
@@ -0,0 +1,191 @@
+import org.junit.rules.TemporaryFolder;
+import org.mockito.InOrder;
+import org.mockito.Mockito;
+
+public class AsyncSnapshotingTest {
+
+  private final TemporaryFolder tempFolderRule = new TemporaryFolder();
+  private final AutoCloseableRule autoCloseableRule = new AutoCloseableRule();
+  private final LogStreamRule logStreamRule = new LogStreamRule(tempFolderRule);
+
+  @Rule
+  public final RuleChain chain =
+      RuleChain.outerRule(autoCloseableRule).around(tempFolderRule).around(logStreamRule);
+
+  private StateSnapshotController snapshotController;
+  private LogStream logStream;
+  private AsyncSnapshotDirector asyncSnapshotDirector;
+
+  @Before
+  public void setup() throws IOException {
+    final File snapshotsDirectory = tempFolderRule.newFolder("snapshots");
+    final File runtimeDirectory = tempFolderRule.newFolder("runtime");
+    final StateStorage storage = new StateStorage(runtimeDirectory, snapshotsDirectory);
+
+    snapshotController =
+        new StateSnapshotController(
+            ZeebeRocksDbFactory.newFactory(DefaultColumnFamily.class), storage);
+    snapshotController.openDb();
+    autoCloseableRule.manage(snapshotController);
+    snapshotController = spy(snapshotController);
+
+    logStreamRule.setCommitPosition(25L);
+    logStream = spy(logStreamRule.getLogStream());
+    final ActorScheduler actorScheduler = logStreamRule.getActorScheduler();
+
+    final Supplier<ActorFuture<Long>> writtenSupplier = mock(Supplier.class);
+    when(writtenSupplier.get())
+        .thenReturn(CompletableActorFuture.completed(99L), CompletableActorFuture.completed(100L));
+
+    asyncSnapshotDirector =
+        new AsyncSnapshotDirector(
+            "processor-1",
+            writtenSupplier,
+            snapshotController,
+            actorCondition -> logStream.registerOnCommitPositionUpdatedCondition(actorCondition),
+            actorCondition -> logStream.removeOnCommitPositionUpdatedCondition(actorCondition),
+            () -> logStream.getTerm(),
+            () -> logStream.getCommitPosition());
+    actorScheduler.submitActor(asyncSnapshotDirector).join();
+  }
+
+  @Test
+  public void shouldStartToTakeSnapshot() {
+    // given
+
+    // when
+    logStreamRule.getClock().addTime(Duration.ofMinutes(1));
+
+    // then
+    final InOrder inOrder = Mockito.inOrder(snapshotController, logStream);
+    inOrder
+        .verify(logStream, timeout(500L).times(1))
+        .registerOnCommitPositionUpdatedCondition(any());
+    inOrder.verify(snapshotController, timeout(500L).times(1)).takeTempSnapshot();
+    inOrder.verify(logStream, timeout(500L).times(1)).getCommitPosition();
+    inOrder.verifyNoMoreInteractions();
+  }
+
+  @Test
+  public void shouldWaitUntilCommitPositionWasUpdated() throws Exception {
+    // given
+    logStreamRule.getClock().addTime(Duration.ofMinutes(1));
+
+    // when
+    waitUntil(() -> asyncSnapshotDirector.lastWrittenEventPosition != INITIAL_POSITION);
```

Comment:
Please avoid using the internal state of the director. This test should also work without this check.